src/fsharp/FSharp.Core/iseq.fs

```diff
@@ -0,0 +1,1345 @@
+                upcast { new FolderWithPostProcessing<'T,unit,IEnumerator<'U>> ((),source2.GetEnumerator()) with
+                    override this.ProcessNext value =
+                        if this.State.MoveNext() then
+                            f value this.State.Current
+                        else
+                            (this :> IOutOfBand).StopFurtherProcessing pipeIdx
+                        Unchecked.defaultof<_> (* return value unused in Fold context *)
+
+                    override this.OnComplete _ = ()
+                    override this.OnDispose () = this.State.Dispose () })
+
+        [<CompiledName "IterateIndexed2">]
+        let inline iteri2 (f:int->'T->'U->unit) (source1:ISeq<'T>) (source2:ISeq<'U>) : unit =
+            source1.Fold (fun pipeIdx ->
+                upcast { new FolderWithPostProcessing<'T,unit,Values<int,IEnumerator<'U>>>((),Values<_,_>(0,source2.GetEnumerator())) with
+                    override this.ProcessNext value =
+                        if this.State._2.MoveNext() then
+                            f this.State._1 value this.State._2.Current
+                            this.State._1 <- this.State._1 + 1
+                            Unchecked.defaultof<_>
+                        else
+                            (this :> IOutOfBand).StopFurtherProcessing pipeIdx
+                            Unchecked.defaultof<_>
+                    override this.OnComplete _ = () 
+                    override this.OnDispose () = this.State._2.Dispose () })
+
+        [<CompiledName "TryHead">]
+        let tryHead (source:ISeq<'T>) =
+            source.Fold (fun pipeIdx ->
+                { new Folder<'T, Option<'T>> (None) with
+                    override this.ProcessNext value =
+                        this.Result <- Some value
+                        (this :> IOutOfBand).StopFurtherProcessing pipeIdx
+                        Unchecked.defaultof<_> (* return value unused in Fold context *) })
+
+        [<CompiledName "Head">]
+        let head (source:ISeq<_>) =
+            match tryHead source with
+            | None -> invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
+            | Some x -> x
+
+        [<CompiledName "IterateIndexed">]
+        let inline iteri f (source:ISeq<'T>) =
+            source.Fold (fun _ ->
+                upcast { new FolderWithState<'T,unit,int> ((),0) with
+                    override this.ProcessNext value =
+                        f this.State value
+                        this.State <- this.State + 1
+                        Unchecked.defaultof<_> (* return value unused in Fold context *) })
+
+        [<CompiledName "Except">]
+        let inline except (itemsToExclude: seq<'T>) (source:ISeq<'T>) : ISeq<'T> when 'T:equality =
+            checkNonNull "itemsToExclude" itemsToExclude
+            source.PushTransform { new ITransformFactory<'T,'T> with
+                override __.Compose _ _ next =
```

Comment:
Please don't use curried arguments for ``Compose``.  Just use three uncurried arguments.